tensorflow/lite/nnapi/nnapi_implementation.cc

```diff
@@ -65,21 +65,13 @@ void* LoadFunction(void* handle, const char* name, bool optional) {
 
 #ifndef __ANDROID__
 // Add /dev/shm implementation of shared memory for non-Android platforms
-int ASharedMemory_create(const char* /* name */, size_t size) {
+int ASharedMemory_create(const char* name, size_t size) {
   // Each call to ASharedMemory_create produces a unique memory space, hence
-  // name should not be used to create the shared memory file, otherwise
+  // name should not be unique, otherwise
   // two calls to create memory regions using the same 'name', will collide.
-  char shm_name_buffer[L_tmpnam];
-  if (tmpnam(shm_name_buffer) == nullptr) {
-    return -1;
-  }
-
-  // tmpnam will produce a string containing with slashes, but shm_open
-  // won't like that.
-  std::string shm_region_name = std::string(shm_name_buffer);
-  std::replace(shm_region_name.begin(), shm_region_name.end(), '/', '-');
+  // Caller is responsible to provide a unique name.
 
-  int fd = shm_open(shm_region_name.c_str(), O_RDWR | O_CREAT, 0644);
+  int fd = shm_open(name, O_RDWR | O_CREAT | O_EXCL, 0644);
```

Comment:
maybe worth adding a comment on using O_CREAT | O_EXCL together to not clobber an existing file?

Reply:
Thanks, good point.